tests/rados/test_bluestore_features.py

```diff
@@ -0,0 +1,169 @@
+    # Executing tests for 45 minutes
+    time_execution = datetime.datetime.now() + datetime.timedelta(minutes=45)
+    while datetime.datetime.now() < time_execution:
+        # Get all OSDs heap dump
+        heap_dump = rados_object.get_heap_dump(osd_list)
+        # get the osd application used memory
+        osd_app_mem = get_bytes_used_by_app(heap_dump)
+        total_osd_app_mem = mergeDictionary(total_osd_app_mem, osd_app_mem)
+        # wait for 10 seconds and collecting the memory
+        time.sleep(10)
+    for osd_id, mem_list in total_osd_app_mem.items():
+        mem_growth = is_what_percent_mem(mem_list)
+        if mem_growth > 80:
+            log.error(
+                f"The osd.{osd_id} consuming more memory with the relative memory growth {mem_growth}"
+            )
+            raise Exception("No warning generated by PG Autoscaler")
+        log.info(f"The relative memory growth for the osd.{osd_id} is {mem_growth} ")
+
+    rados_object.change_heap_profiler_state(osd_list, "stop")
+
+    # check fo the crashes in the cluster
+    crash_list = rados_object.do_crash_ls()
+    if not crash_list:
+        return 0
+    else:
+        return 1
+
+
+def is_what_percent_mem(mem_list):
+    """
+    To calculate the relative memory growth
+    Args:
+         mem_list: list of mllac values of the application
+    Return:
+          result : Finds the relative growth of min amd max values
+
+    """
+    min_mem = int(min(mem_list))
+    max_mem = int(max(mem_list))
+    result = float(((max_mem - min_mem) * 100) / min_mem)
+    return result
+
+
+def get_bytes_used_by_app(osds_heap_dump):
+    """
+    To gets the bytes ued by the application in the heap dump
+    Args:
+         osds_heap_dump: List of osd heap dumps
+         Example of heap dump:
+              MALLOC:      429538736 (  409.6 MiB) Bytes in use by application
+              MALLOC: +            0 (    0.0 MiB) Bytes in page heap freelist
+              MALLOC: +      8934864 (    8.5 MiB) Bytes in central cache freelist
+              MALLOC: +      3876864 (    3.7 MiB) Bytes in transfer cache freelist
+              MALLOC: +     20636800 (   19.7 MiB) Bytes in thread cache freelists
+
+    Return:
+          osd_id_app_mem: Returns a dictionary with OSD id as a key and
+                          malloc for the OSD application
+                   Example: {1:['429538736' , '456789',..], 2:['21312','76576',..]...}
+    """
+    osd_id_app_mem = {}
+    app_str = "Bytes in use by application"
+    for osd_id, osd_dump in osds_heap_dump.items():
+        malloc_str = osd_dump.split("MALLOC:")
+        result = [app_dump for app_dump in malloc_str if app_str in app_dump]
+        result = (" ".join(result)).strip()
+        result_list = result.split("(")
+        osd_id_app_mem[osd_id] = result_list[0].strip()
+    return osd_id_app_mem
+
+
+# Merge dictionaries
+def mergeDictionary(dict_1, dict_2):
```

Comment:
can we use built-in method for this? dict1.update(dict2) ? The return would be exactly same

Reply:
With the update, the new value of the key will get replaced with the old value. Here I am appending the new memory value to the key.

Reply:
Please update the docstrings appropriately, adding summary, usahe, examples and return types with examples.

Reply:
Doc string is added